source/ConditionSet.cpp

```diff
@@ -23,203 +23,103 @@ this program. If not, see <https://www.gnu.org/licenses/>.
-		for(const string &str : tokens)
-			if(invalids.contains(str))
-				return true;
-		return false;
-	}
-
-	// Ensure the ConditionSet line has balanced parentheses on both sides.
-	bool HasUnbalancedParentheses(const vector<string> &tokens)
-	{
-		int parentheses = 0;
-		for(const string &str : tokens)
+		switch(op)
 		{
-			if(parentheses < 0)
-				return true;
-			else if(parentheses && (IsAssignment(str) || IsComparison(str)))
-				return true;
-			else if(str == "(")
-				++parentheses;
-			else if(str == ")")
-				--parentheses;
-		}
-		return parentheses;
-	}
-
-	// Perform a preliminary assessment of the input condition, to determine if it is remotely well-formed.
-	// The final assessment of its validity will be whether it parses into an evaluable Expression.
-	bool IsValidCondition(const DataNode &node)
-	{
-		const vector<string> &tokens = node.Tokens();
-		int assigns = count_if(tokens.begin(), tokens.end(), IsAssignment);
-		int compares = count_if(tokens.begin(), tokens.end(), IsComparison);
-		if(assigns + compares != 1)
-			node.PrintTrace("Error: An expression must either perform a comparison or assign a value:");
-		else if(HasInvalidOperators(tokens))
-			node.PrintTrace("Error: Brackets, braces, exponentiation, and boolean/bitwise math are not supported:");
-		else if(HasUnbalancedParentheses(tokens))
-			node.PrintTrace("Error: Unbalanced parentheses in condition expression:");
-		else if(count_if(tokens.begin(), tokens.end(), [](const string &token)
-				{ return token.size() > 1 && token.front() == '('; }))
-			node.PrintTrace("Error: Parentheses must be separate from tokens:");
-		else
-			return true;
-
-		return false;
-	}
-
-	// Converts the given vector of condition tokens (like "reputation: Republic",
-	// "random", or "4") into the integral values they have at runtime.
-	vector<int64_t> SubstituteValues(const vector<string> &side, const ConditionsStore &conditions)
-	{
-		auto result = vector<int64_t>();
-		result.reserve(side.size());
-		for(const string &str : side)
-		{
-			int64_t value = 0;
-			if(DataNode::IsNumber(str))
-				value = static_cast<int64_t>(DataNode::Value(str));
-			else
-			{
-				const auto perm = conditions.Get(str);
-				if(perm)
-					value = perm;
-			}
-			result.emplace_back(value);
+			case ConditionSet::ExpressionOp::OP_INVALID:
+				return 9;
+			case ConditionSet::ExpressionOp::OP_LIT:
+			case ConditionSet::ExpressionOp::OP_VAR:
+				return 8;
+			case ConditionSet::ExpressionOp::OP_MUL:
+			case ConditionSet::ExpressionOp::OP_DIV:
+			case ConditionSet::ExpressionOp::OP_MOD:
+				return 6;
+			case ConditionSet::ExpressionOp::OP_ADD:
+			case ConditionSet::ExpressionOp::OP_SUB:
+				return 5;
+			case ConditionSet::ExpressionOp::OP_EQ:
+			case ConditionSet::ExpressionOp::OP_NE:
+			case ConditionSet::ExpressionOp::OP_GT:
+			case ConditionSet::ExpressionOp::OP_LT:
+			case ConditionSet::ExpressionOp::OP_GE:
+			case ConditionSet::ExpressionOp::OP_LE:
+				return 3;
+			default:
```

Comment:
You should be able to return here if ExpressionOp is an enum class.

Reply:
Compiler agreed. I will put the return in the default here.